credential.go

```diff
@@ -0,0 +1,84 @@
+package mls
+
+import (
+	"fmt"
+	"github.com/bifurcation/mint/syntax"
+)
+
+type CredentialType uint8
+
+const (
+	credential_basic = iota
```

Comment:
More idiomatic to recite the enum type in the values, so `CredentialTypeBasic`, `CredentialTypeX509`.

Also, for values that need to be interoperable, please specify values explicitly instead of using `iota`.